### DISCUSSION TITLE
How to confirm presence of ST7567 on I2C

### DISCUSSION THREAD

**Discussion by strud** [olikraus/u8g2#2494] [Q&A] [answered]
I have a hardware configuration that may or may not have an ST7567 display connected. Therefore I want to check it's presence over I2C but I cannot find a library function in U8g2 that returns a bool or similar to indicate successful communication or presence.

Is there anything built in the library to allow this?

**Comment by olikraus** [comment, marked as answer]
This is out fo the scope of u8g2 library. You can do this with the standard Arduino library.

1. Send the I2C address of the ST7567 with https://www.arduino.cc/reference/en/language/functions/communication/wire/begintransmission/
2. Then listen with https://www.arduino.cc/reference/en/language/functions/communication/wire/endtransmission/ for the response.

if you receive anything else than 0 with the second step, then you can assume, that the display is not connected

**Comment by strud** [comment]
Thanks, will implement this.
